common/workunit/workunit.cpp

```diff
@@ -8012,17 +8091,37 @@ IConstWUStatisticIterator& CLocalWorkUnit::getStatistics(const IStatisticsFilter
     return * new CCompoundIteratorOf<IConstWUStatisticIterator, IConstWUStatistic>(localStats, graphStats);
 }
 
-IConstWUStatistic * CLocalWorkUnit::getStatistic(const char * creator, const char * scope, StatisticKind kind) const
+IConstWUStatistic * CLocalWorkUnit::getStatistic(const char * scope, StatisticKind kind) const
 {
+#if 0
+    //MORE: Optimize this....
+    WuScopeFilter filter;
+    filter.addScope(scope).setIncludeNesting(0).finishedFilter();
+    Owned<IConstWUScopeIterator> stats = &getScopeIterator(&filter);
+    if (stats->first())
+        return stats->getStat(kind, value);
+    return NULL;
+#else
     //MORE: Optimize this....
     StatisticsFilter filter;
-    filter.setCreator(creator);
     filter.setScope(scope);
     filter.setKind(kind);
     Owned<IConstWUStatisticIterator> stats = &getStatistics(&filter);
     if (stats->first())
         return LINK(&stats->query());
     return NULL;
+#endif
+}
+
+bool CLocalWorkUnit::getStatistic(stat_type & value, const char * scope, StatisticKind kind) const
+{
+    //MORE: Optimize this....
+    WuScopeFilter filter;
+    filter.addScope(scope).setIncludeNesting(0).addRequiredStat(kind).addOutputStatistic(kind).finishedFilter();
```

Comment:
@ghalliday Shouldn't addOutputStatistic automatically addRequiredStat?

Reply:
No - you may want to know about a particular stat if it exists, but not require it to be present.